jpa/boot-data-customsequence/src/main/java/com/example/custom/sequence/mapper/CustomerMapper.java

```diff
@@ -1,12 +1,80 @@
 package com.example.custom.sequence.mapper;
 
 import com.example.custom.sequence.entities.Customer;
+import com.example.custom.sequence.entities.Order;
+import com.example.custom.sequence.model.request.CustomerRequest;
 import com.example.custom.sequence.model.response.CustomerResponse;
+import com.example.custom.sequence.repositories.OrderRepository;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
 import org.springframework.stereotype.Service;
 
 @Service
 public class CustomerMapper {
+
+    private final OrderMapper orderMapper;
+    private final OrderRepository orderRepository;
+
+    public CustomerMapper(OrderMapper orderMapper, OrderRepository orderRepository) {
+        this.orderMapper = orderMapper;
+        this.orderRepository = orderRepository;
+    }
+
     public CustomerResponse mapToResponse(Customer saved) {
-        return new CustomerResponse(saved.getId(), saved.getText());
+        return new CustomerResponse(
+                saved.getId(),
+                saved.getText(),
+                orderMapper.mapToResponseListWithOutCustomer(saved.getOrders()));
+    }
+
+    public Customer mapToEntity(CustomerRequest customerRequest) {
+        Customer customer = new Customer(customerRequest.text());
+        customerRequest
+                .orders()
+                .forEach(orderRequest -> customer.addOrder(orderMapper.mapToEntity(orderRequest)));
+        return customer;
+    }
+
+    public void updateCustomerFromRequest(CustomerRequest customerRequest, Customer foundCustomer) {
+        foundCustomer.setText(customerRequest.text());
+        List<Order> removedOrders = new ArrayList<>(foundCustomer.getOrders());
+        List<Order> ordersFromRequest =
+                customerRequest.orders().stream()
+                        .map(
+                                orderRequest ->
+                                        orderMapper.mapToEntityWithCustomer(
+                                                orderRequest, foundCustomer))
+                        .collect(Collectors.toList());
+        removedOrders.removeAll(ordersFromRequest);
+
+        for (Order removedOrder : removedOrders) {
```

Comment:
@coderabbitai, please write integrationTests in `CustomerControllerIT` to showcase all 3 scenarions as per the code

Reply:
@coderabbitai, In above examples include various orderRequests like (newer one, updated one, delete ones )as of now all are empty